node_tree.py

```diff
@@ -116,6 +146,71 @@ def replace_socket(self, new_type, new_name=None):
         return the new socket, the old reference might be invalid"""
         return replace_socket(self, new_type, new_name)
 
+    @property
+    def extra_info(self):
+        # print("getting base extra info")
+        return ""
+
+    def draw_expander_template(self, context, layout, prop_origin, prop_name="prop"):
+        if self.use_expander and self.bl_idname != "StringsSocket":
```

Comment:
This could be further optimized by doing a test for StringSocket once:
```python
        if self.bl_idname == "StringsSocket":
            layout.prop(prop_origin, prop_name)
        else: # all the other socket types
            if self.use_expander:
                split = layout.split(percentage=.2, align=True)
                c1 = split.column(align=True)
                c2 = split.column(align=True)
                if self.expanded:
                    c1.prop(self, "expanded", icon='TRIA_UP', text='')
                    c1.label(text=self.name[0])
                    c2.prop(prop_origin, prop_name, text="", expand=True)
                else:  # collapsed
                    c1.prop(self, "expanded", icon='TRIA_DOWN', text="")
                    row = c2.row(align=True)
                    if self.bl_idname == "SvColorSocket":
                        row.prop(prop_origin, prop_name)
                    else:
                        row.template_component_menu(prop_origin, prop_name, name=self.name)
            else: # not using expander
                layout.template_component_menu(prop_origin, prop_name, name=self.name)

```